src/main/java/seedu/address/model/BorrowerRecords.java

```diff
@@ -40,7 +40,9 @@ public BorrowerRecords(ReadOnlyBorrowerRecords toBeCopied) {
      * @param borrower Borrower to be checked.
      */
     public boolean hasBorrower(Borrower borrower) {
-        return borrowersMap.containsKey(borrower.getBorrowerId());
+        return listOfBorrowers.stream().anyMatch(current -> current.getName().equals(borrower.getName())
```

Comment:
I think we shouldn't check for same name, just check:
1) if phone number exist, error message is that phone number is already in use
2) if email exits, error message is that email is already in use

With your current implementation, if a user uses a new name but a used phone/email, it will allow borrower to be registered, which shouldn't be the case

Reply:
Noted. Will do the changes!